src/key.js

```diff
@@ -106,17 +113,17 @@ class Key {
    * @returns {void}
    */
   clean () {
-    if (!this._buf || this._buf.length === 0) {
-      this._buf = Buffer.from(pathSepS)
+    if (!this._buf || this._buf.byteLength === 0) {
+      this._buf = pathSepB
     }
 
     if (this._buf[0] !== pathSep) {
-      this._buf = Buffer.concat([pathSepB, this._buf])
+      this._buf = Uint8Array.of(pathSep, ...this._buf)
```

Comment:
📝 I do not know how large `...this._buf` can get, but JS engines (at least sure spidermonkey) have a cap on number of arguments function can take. If it can get arbitrary long it would be better to do
`Uint8Array.from([pathSep, ...this._buf])` instead or this if this is on hot path:

```js
const bytes = new Uint8Array(this._buf.byteLength + 1)
bytes.fill(pathSep, 0, 1)
bytes.set(this._buf, 1)
this._buf = bytes
```

Reply:
The second version is probably better as you avoid creating the intermediate array.